index.js

```diff
@@ -116,30 +126,44 @@ function applyAttributes(t, openingElement, componentName, sourceFileName, compo
   }
   if (!openingElement.node.attributes) openingElement.node.attributes = {}
 
+  const elementName = openingElement.node.name.name || 'unknown'
+  const ignoredComponentFromOptions = ignoreComponentsFromOption && ignoreComponentsFromOption.find(component => {
+    let match = true;
+    if (component[0] !== '*' && component[0] !== sourceFileName) match = false;
+    else if (component[1] !== '*' && component[1] !== componentName) match = false;
+    else if (component[2] !== '*' && component[2] !== elementName) match = false;
+    
+    return match
+  }) !== undefined
+
   let ignoredElement = false
   // Add a stable attribute for the element name but only for non-DOM names
-  if(openingElement.node.attributes.find(node => {
+  if (
+    !ignoredComponentFromOptions
+    && openingElement.node.attributes.find(node => {
     if (!node.name) return
     return node.name.name === elementAttributeName
-  }) == null){
-    const name = openingElement.node.name.name || 'unknown'
-    if (ignoredElements.includes(name)) {
+  }) == undefined){
+    if (defaultIgnoredElements.includes(elementName)) {
       ignoredElement = true
     } else {
       openingElement.node.attributes.push(
         t.jSXAttribute(
           t.jSXIdentifier(elementAttributeName),
-          t.stringLiteral(name)
+          t.stringLiteral(elementName)
         )
       )
     }
   }
 
   // Add a stable attribute for the component name (absent for non-root elements)
-  if(componentName && (openingElement.node.attributes.find(node => {
-    if (!node.name) return
-    return node.name.name === componentAttributeName
-  }) == null)){
+  if (
+    componentName 
+    && !ignoredComponentFromOptions
+    && (openingElement.node.attributes.find(node => {
+      if (!node.name) return
+      return node.name.name === componentAttributeName
+    }) == undefined)){
```

Comment:
Maybe we could incapsulate this logic into a `hasNodeNamed(openingElement, name)` helper?

Reply:
Thanks for the recommendations! I updated according to your comments, mind giving this another look through?